### PR TITLE
Fix Retry Seconds Milliseconds Confusion

### PR DESCRIPTION
**Pull request by anthony-murphy**
Fix bug in retry logic where we were not consistent with time units. specifically sometime we used seconds, and other time we use milliseconds, which lead to incorrect timeout delays

### CODE REVIEW

**Review by jatgarg — APPROVED** [review]

**Comment by jatgarg** [comment]
I think you will need to fix a test case too in retryUtils.spec.ts and convert milliseconds to seconds.

File: packages/loader/container-loader/src/utils.ts [hunk 1]
```diff
@@ -152,12 +152,12 @@ export async function runWithRetry<T>(
             lastError = err;
             // If the error is throttling error, then wait for the specified time before retrying.
             // If the waitTime is not specified, then we start with retrying immediately to max of 8s.
-            retryAfter = getRetryDelayFromError(err) ?? Math.min(retryAfter * 2, 8000);
```
Comment (vladsud) [inline packages/loader/container-loader/src/utils.ts:155]: Personally, I'd prefer all the times to be milliseconds everywhere, and rather rename getRetryDelayFromError and make it return milliseconds (and fix couple places).

Note - same problem exists in getFileLink.

File: packages/test/test-end-to-end-tests/src/test/detachedContainerTests.spec.ts [hunk 2]
```diff
@@ -607,7 +607,7 @@ describeNoCompat("Detached Container", (getTestObjectProvider) => {
         loader = provider.makeTestLoader(testContainerConfig) as Loader;
     });
 
-    it("Retry attaching detached container", async () => {
+    it.only("Retry attaching detached container", async () => {
```
Comment (vladsud) [inline packages/test/test-end-to-end-tests/src/test/detachedContainerTests.spec.ts:610]: I assume this needs to be reverted

File: packages/loader/container-loader/src/test/retryUtils.spec.ts [hunk 3]
```diff
@@ -8,13 +8,24 @@ import { DriverErrorType } from "@fluidframework/driver-definitions";
...
+    (callback: (...cbArgs: any[]) => void, ms: number, ...args: any[]) => _setTimeout(callback, ms / 1000.0, ...args);
+
 describe("Retry Util Tests", () => {
     const deltaManager = {
         refreshDelayInfo: () => {},
         emitDelayInfo: () => {},
     };
     const logger = new TelemetryNullLogger();
 
+    before(()=>{
```
Comment (vladsud) [inline packages/loader/container-loader/src/test/retryUtils.spec.ts:22]: I'd rather use workflow similar to mockFetchOk / mockFetchMultiple, i.e. replace it for duration of the wrapped callback.
That way it's a bit more cleaner in terms of scope, and we can (and likely should) do more validation, like how many calls are actually changed (see assert(cb !== undefined, "the end") & assert(responses.length === 0, "all responses used") asserts)

**Comment by skylerjokiel** [comment]
Please ensure we are providing descriptions, or links to issues with descriptions in our PRs to make them more inclusive.

File: packages/loader/container-loader/src/utils.ts (hunk 1 above)
Comment (anthony-murphy) [reply packages/loader/container-loader/src/utils.ts:155]: this is just a bug fix, rather not couple it with refactoring. added an issue: #5881

**Pull request merged**